src/rmarkdown/preview.ts

```diff
@@ -0,0 +1,416 @@
+
+import * as cp from 'child_process';
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import * as cheerio from 'cheerio';
+
+import path = require('path');
+import crypto = require('crypto');
+
+
+import { config, doWithProgress, getRpath, readContent, setContext, escapeHtml } from '../util';
+import { extensionContext, tmpDir } from '../extension';
+
+class RMarkdownPreview extends vscode.Disposable {
+    title: string;
+    cp: cp.ChildProcessWithoutNullStreams;
+    panel: vscode.WebviewPanel;
+    resourceViewColumn: vscode.ViewColumn;
+    outputUri: vscode.Uri;
+    htmlDarkContent: string;
+    htmlLightContent: string;
+    fileWatcher: fs.FSWatcher;
+    autoRefresh: boolean;
+
+    constructor(title: string, cp: cp.ChildProcessWithoutNullStreams, panel: vscode.WebviewPanel,
+        resourceViewColumn: vscode.ViewColumn, outputUri: vscode.Uri, uri: vscode.Uri,
+        RMarkdownPreviewManager: RMarkdownPreviewManager, themeBool: boolean, autoRefresh: boolean) {
+        super(() => {
+            this.cp?.kill('SIGKILL');
+            this.panel?.dispose();
+            this.fileWatcher?.close();
+            fs.removeSync(this.outputUri.path);
```

Comment:
Also applies to some other places: `vscode.Uri` has a `fsPath` property which is more robust regarding windows drive letters, backslashes etc.